### PR TITLE
chore: Add tests for ariakit utils/misc

### PR DESCRIPTION
**Pull request by SCasarotto**
More tests for Utils

### CODE REVIEW

File: packages/ariakit-utils/src/__tests__/misc-test.ts [hunk 1]
```diff
@@ -0,0 +1,189 @@
+import { sleep } from "ariakit-test-utils";
+
+import {
```
Comment (diegohaz) [inline packages/ariakit-utils/src/__tests__/misc-test.ts]: Nit:
```suggestion
import { sleep } from "ariakit-test-utils";
import {
```

File: packages/ariakit-utils/src/__tests__/misc-test.ts [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+  queueMicrotask(() => {
+    value = 1;
+  });
+
+  // Advance time enough to make sure the settimout has fired
+  jest.advanceTimersByTime(1000);
+
+  expect(value).toBe(1);
+
+  jest.useRealTimers();
```
Comment (diegohaz) [inline packages/ariakit-utils/src/__tests__/misc-test.ts]: Have you tried with sleep too? Also, I think we should test the value before as well:
```suggestion

  queueMicrotask(() => {
    value = 1;
  });
  expect(value).toBe(0);
  await sleep();
  expect(value).toBe(1);
```

File: packages/ariakit-utils/src/__tests__/misc-test.ts [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+  Object.defineProperty(window, "queueMicrotask", {
+    value: undefined,
+    configurable: true,
+  });
+
+  queueMicrotask(() => {
+    value = 2;
+  });
+  await sleep();
+  expect(value).toBe(2);
```
Comment (diegohaz) [inline packages/ariakit-utils/src/__tests__/misc-test.ts]:
```suggestion
  queueMicrotask(() => {
    value = 2;
  });
  expect(value).toBe(1);
  await sleep();
  expect(value).toBe(2);
```

**Review by diegohaz — COMMENTED** [review]
This looks really good! Thanks! I made a few suggestions.

File: packages/ariakit-utils/src/__tests__/misc-test.ts (hunk 2 above)
Comment (SCasarotto) [reply packages/ariakit-utils/src/__tests__/misc-test.ts]: Good call.

**Review by diegohaz — APPROVED** [review]
Nice! 🚀

**Pull request merged**